src/osd/osd_types.h

```diff
@@ -6229,6 +6229,7 @@ static const __u8 pg_compat_struct_v = 10;
 int prepare_info_keymap(
   CephContext* cct,
   map<string,bufferlist> *km,
+  set<string> *kd,
```

Comment:
i think `std::string *` would suffice. and we could use a better name than "kd", like `key_to_remove` or `stale_key` ?